zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobBatchActivateProcessor.java

```diff
@@ -130,8 +132,10 @@ private void rejectCommand(final TypedRecord<JobBatchRecord> record) {
       rejectionType = RejectionType.INVALID_ARGUMENT;
       rejectionReason = String.format(format, "type", "present", "blank");
     } else {
-      rejectionType = RejectionType.NOT_FOUND;
-      rejectionReason = "No jobs found to activate";
+      rejectionType = RejectionType.UNAUTHORIZED;
+      rejectionReason =
+          "User is not authorized to activate jobs for tenants '%s'"
+              .formatted(value.getTenantIds());
```

Comment:
❌ I'm going to be annoying and refer to the [error message guidelines](https://github.com/camunda/camunda/wiki/Error-Guidelines).

```suggestion
      rejectionType = RejectionType.UNAUTHORIZED;
      rejectionReason =
          "Expected to activate job batch for tenants '%s', but user is not authorized. Authorized tenants are '%s'"
              .formatted(value.getTenantIds(), authorizedTenantIds);
```

It does mean we need to have access to the authorized tenants here. Honestly it's probably easier to do this check in a separate method returning an `Either<Rejection, JobBatchRecord>` and changing `isValid` to be the same. That way you can just chain the validations together.